### PR TITLE
Add more logging when failing watch history entry fails.

### PR DESCRIPTION
**Pull request by martijnvg**
Relates to #30777

### CODE REVIEW

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/2

File: x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java [hunk 1]
```diff
@@ -305,39 +305,58 @@ private ObjectPath getWatchHistoryEntry(String watchId) throws Exception {
...
+                    objectPathReference.set(objectPath);
+                } catch (ResponseException e) {
+                    final String err = "Failed to perform search of watcher history";
+                    logger.info(err, e);
+                    fail(err);
+                }
+            });
+        } catch (AssertionError ae) {
+            {
+                Request request = new Request("GET", "/_watcher/stats");
```
Comment (jakelandis) [inline x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java]: you may want add `metric=_all` to include current and queued watches.
```
/_watcher/stats?metric=_all
```

File: x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java [hunk 2]
```diff
@@ -305,39 +305,58 @@ private ObjectPath getWatchHistoryEntry(String watchId) throws Exception {
...
+                logger.info("watcher_stats: {}", EntityUtils.toString(response.getEntity()));
             }
-        });
+            {
+                Request request = new Request("GET", "/_cluster/state");
+                Response response = client().performRequest(request);
+                logger.info("cluster_state: {}", EntityUtils.toString(response.getEntity()));
+            }
+            {
+                Request request = new Request("GET", "/.watcher-history-*/_search");
```
Comment (jakelandis) [inline x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java]: When this fails and is stuck for during the assertBusy, may result in alot of .watcher_history... should we increase the default size ?

**Comment by jakelandis** [comment]
LGTM - a couple minor suggestions.

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**